sdk/kiosk/src/query/kiosk.ts

```diff
@@ -40,17 +42,20 @@ export type KioskItem = {
   itemId: string;
   /** The type of the Item */
   itemType: string;
+  /** Whether the item is Locked (there must be a `Lock` Dynamic Field) */
+  isLocked: boolean;
+  /** Optional listing */
+  listing?: KioskListing;
```

Comment:
This is tricky. I would suggest unwrapping it into the first layer. We always know that a listing exists and whether it is exclusive. One thing we don't know is the `price`.